### PR TITLE
Swiot tutorial

### PR DESCRIPTION
**Pull request by Andrei-Fabian-Pop**
TutorialBuilder implemetation in SWIOT plugin.

### CODE REVIEW

File: plugins/swiot/src/config/configtutorial.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+		// add all children from the current object
+		for (auto child : children) {
+			stack.push(child);
+		}
+
+		QString currentObjectName = currentObject->property("tutorial_name").toString();
+
+		// check weather the object fits the search
+		if (m_tutorialObjectNames.contains(currentObjectName)) {
```
Comment (Andrei-Fabian-Pop) [inline plugins/swiot/src/config/configtutorial.cpp:61]: refactor

File: plugins/swiot/src/config/configtutorial.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	/**
+	 * @brief Collects and adds all qt objects that are part of the tutorial and are contained in the config widget
+	 * (given as constructor parameter) widget recursively in the chapter list. The widgets are considered part of
+	 * the tutorial if their object name is in the CONFIG_TUTORIAL_JSON file under the "config" section. After
+	 * they are collected, they are sorted based on the number read from the json file and added as chapters. The
+	 * chapter description will be matched and added as well from the same json file.
+	 * @return The number of objects collected.
+	 * */
+	unsigned int collectChapters();
```
Comment (Andrei-Fabian-Pop) [inline plugins/swiot/src/config/configtutorial.h:48]: change to uint16_t

File: plugins/swiot/src/config/configtutorial.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	/**
+	 * @brief Collects the index, name, description and populates m_tutorialObjectNames and m_chapters with the
+	 * specified attributes
+	 * */
+	void readTutorialRequirements();
+
+private:
+	QWidget* m_config;
+
+	struct ChapterInstructions {
```
Comment (Andrei-Fabian-Pop) [inline plugins/swiot/src/config/configtutorial.h:59]: review this after changing requirements collection algo

File: plugins/swiot/src/swiotinfopage.cpp [hunk 4]
```diff
@@ -29,19 +29,24 @@ SwiotInfoPage::SwiotInfoPage(QWidget* parent) : InfoPage(parent)
 {
 	// not enough attributes for correct display with normal size policies, there is too much spacing without this
 	this->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
+	this->initTempCheckBox();
+	this->layout()->addWidget(m_enTempReadCheckBox);
+}
+
+void SwiotInfoPage::enableTemperatureReadBtn(bool enable)
```
Comment (Andrei-Fabian-Pop) [inline plugins/swiot/src/swiotinfopage.cpp:36]: investigate temp checkbox behavior based on connected context

**Review by bindea-cristian — APPROVED** [review]

**Review by AlexandraTrifan — APPROVED** [review]

**Pull request merged**